src/ipv4.c

```diff
@@ -400,28 +402,30 @@ int ipv4_restore_routes(struct tunnel *tunnel)
 		if (ret != 0)
 			log_warn("Could not delete route to vpn server (%s).\n",
 			         err_ipv4_str(ret));
-	} else {
-		log_debug("Route to vpn server is not added\n");
-	}
 
-	if (tunnel->ipv4.split_routes)
-		goto out;
+		if (tunnel->ipv4.split_routes)
+			goto out;
 
-	ret = ipv4_del_route(ppp_rt);
-	if (ret != 0)
-		log_warn("Could not delete route through tunnel (%s).\n",
-		         err_ipv4_str(ret));
+		ret = ipv4_del_route(ppp_rt);
+		if (ret != 0)
+			log_warn("Could not delete route through tunnel (%s).\n",
+			         err_ipv4_str(ret));
 
-	// Restore the default route
-	// It seems to not be automatically restored on all linux distributions
-	ret = ipv4_set_route(def_rt);
-	if (ret != 0)
-		log_warn("Could not restore default route (%s). Already restored?\n",
-		         err_ipv4_str(ret));
+		// Restore the default route
+		// It seems to not be automatically restored on all linux distributions
+		ret = ipv4_set_route(def_rt);
+		if (ret != 0) {
+			log_warn("Could not restore default route (%s). \
+Already restored?\n",
+			         err_ipv4_str(ret));
+		}
 
-	route_destroy(ppp_rt);
+		route_destroy(ppp_rt);
 out:
-	route_destroy(def_rt);
+		route_destroy(def_rt);
+	} else {
+		log_debug("Route to vpn server is not added\n");
```

Comment:
"...was not added" would be better, because we are in the cleanup routine that is called at the end of the session here